catchup/catchpointService.go

```diff
@@ -282,7 +282,17 @@ func (cs *CatchpointCatchupService) processStageLedgerDownload() (err error) {
 	}
 
 	// download balances file.
-	peerSelector := makePeerSelector(cs.net, []peerClass{{initialRank: peerRankInitialFirstPriority, peerClass: network.PeersPhonebookRelays}})
+	var peerSelector *peerSelector
+	if cs.config.EnableCatchupFromArchiveServers {
+		peerSelector = makePeerSelector(cs.net, []peerClass{
+			{initialRank: peerRankInitialFirstPriority, peerClass: network.PeersPhonebookArchivers},
+			{initialRank: peerRankInitialSecondPriority, peerClass: network.PeersPhonebookRelays},
+		})
+	} else {
+		peerSelector = makePeerSelector(cs.net, []peerClass{
+			{initialRank: peerRankInitialFirstPriority, peerClass: network.PeersPhonebookRelays},
+		})
+	}
```

Comment:
Ignore if too nitpicky: Maybe declare the `[]peerClass` slice with the first element in it, conditionally append to it, then make a single call to `makePeerSelector`.